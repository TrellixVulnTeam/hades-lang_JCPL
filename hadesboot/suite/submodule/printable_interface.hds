import submodule.test_utils as utils;

implement Printable for CInt {
    def print(this: CInt): Void {
        utils.print_int(this);
    }
}

interface Printable {
    def print(this: This): Void;
}