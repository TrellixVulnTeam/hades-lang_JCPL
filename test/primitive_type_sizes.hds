import submodule.test_utils as utils;

def print_size[T](name: *Byte): Void {
    utils.print_byte_string(name);
    utils.print_byte_string(b": ");
    utils.print_size(size_of[T]);
}

def main(): Void {
    print_size[u8](b"u8");
    print_size[s8](b"s8");
    print_size[u16](b"u16");
    print_size[s16](b"s16");
    print_size[u32](b"u32");
    print_size[s32](b"s32");
    print_size[u64](b"u64");
    print_size[s64](b"s64");
    print_size[f16](b"f16");
    print_size[f32](b"f32");
    print_size[f64](b"f64");
}
