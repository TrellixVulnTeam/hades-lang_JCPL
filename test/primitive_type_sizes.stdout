u8: 1
s8: 1
u16: 2
s16: 2
u32: 4
s32: 4
u64: 8
s64: 8
f16: 2
f32: 4
f64: 8
